met-web/src/components/engagement/form/ActionContext.tsx

```diff
@@ -53,13 +58,38 @@ export const ActionProvider = ({ children }: { children: JSX.Element }) => {
     const [loadingSavedEngagement, setLoadingSavedEngagement] = useState(true);
 
     const [savedEngagement, setSavedEngagement] = useState<Engagement>(createDefaultEngagement());
-    const [modalState, setModalState] = useState<EngagementFormModalState>({
-        modalOpen: false,
-    });
 
     const [bannerImage, setBannerImage] = useState<File | null>();
     const [savedBannerImageFileName, setSavedBannerImageFileName] = useState('');
 
+    const [widgets, setWidgets] = useState<WidgetsList[]>([]);
+    const [widgetDrawerOpen, setWidgetDrawerOpen] = useState(false);
+    const [widgetDrawerTabValue, setWidgetDrawerTabValue] = React.useState('widgetOptions');
+
+    useEffect(() => {
+        setWidgets([
+            {
+                widget_type: 1,
```

Comment:
We can have a enum with the widget types so it’s clearer to what 1 means

Reply:
for sure, will do this once I create the widget_type table.